src/nvim/api/private/defs.h

```diff
@@ -22,6 +22,14 @@ typedef enum {
   kErrorTypeValidation
 } ErrorType;
 
+typedef enum {
+  kMessageTypeRequest,
+  kMessageTypeResponse,
+  kMessageTypeNotification
+} MessageType;
+
+#define NO_RESPONSE UINT64_MAX
```

Comment:
Perhaps this could be made into an enum? I've been doing this recently, and it's much nicer to debug IMO.